### PR TITLE
Wrap all imported fs methods to set error stack trace limit to zero during their execution

### PR DESCRIPTION
**Pull request by weswigham**
This is just a followup to https://github.com/microsoft/TypeScript/pull/40043 that blanket applies the idea to all `fs` methods we use (or will use in the future), rather than just one specific `statSync` usage.

### CODE REVIEW

**Comment by weswigham** [comment]
@typescript-bot perf test this
@typescript-bot pack this

File: src/compiler/sys.ts [hunk 1]
```diff
@@ -1125,6 +1125,26 @@ namespace ts {
         return parseInt(version.substring(1, dot));
     }
 
+    function wrapStacktraceReduction(mod: any): any {
+        const result = {} as any;
+        for (const key in mod) {
+            if (!(mod[key] instanceof Function)) continue;
+            result[key] = (...args: any[]) => {
```
Comment (DanielRosenwasser) [inline src/compiler/sys.ts:1132]: Nit: avoid the allocation and just use `arguments`

Comment (DanielRosenwasser) [reply src/compiler/sys.ts:1132]: On second thought, maybe `arguments` allocates too. I dunno, consider just not auto-wrapping `fs`.

Comment (DanielRosenwasser) [inline src/compiler/sys.ts:1132]: Worth understanding whether an allocation for this array in each call is a potential problem.

File: src/compiler/sys.ts [hunk 2]
```diff
@@ -1135,7 +1155,7 @@ namespace ts {
 
         function getNodeSystem(): System {
             const nativePattern = /^native |^\([^)]+\)$|^(internal[\\/]|[a-zA-Z0-9_\s]+(\.js)?$)/;
-            const _fs: typeof import("fs") = require("fs");
+            const _fs: typeof import("fs") = wrapStacktraceReduction(require("fs"));
```
Comment (sheetalkamat) [inline src/compiler/sys.ts:1158]: if we are doing this in sys for all functions, we should do it for all the other uses of fs too.. (nodeTypingInstaller, server etc)

Comment (weswigham) [reply src/compiler/sys.ts:1158]: Hm, fair enough, I just didn't realize other things used `fs` directly rather than `sys`.

**Comment by amcasey** [comment]
Is it really the case that we never want a stack from `fs`?  I was only thinking of applying this to `statSync` calls.  Do you have an example of what a stack will look like if, e.g. a file we're trying to write to is locked?

**Comment by weswigham** [comment]
We never inspect the `.stack` property of an error thrown by these calls, and always `catch` them (and either always do one thing or inspect the error code and change the action depending on it) - so programatically, no, we never care about the stack from these.

**Comment by RyanCavanaugh** [comment]
This doesn't seem like a great win for perf overall. We're introducing this overhead on *every* fs call, even though there's only a small number that regularly throw, and also introducing a duplicative `try` block for the cases that do have a wrapping try/catch. I'd want to see more concrete evidence that this is an overall perf win before introducing this complexity.

**Pull request closed without merging**